1-getting-started/lessons/4-connect-internet/translations/README.bn.md

```diff
@@ -0,0 +1,438 @@
+
+পাঠের এই অংশটিতে আইওটি নাইটলাইটিকে ইন্টারনেটে সংযুক্ত করি যাতে করে সেটিকে দূর থেকে নিয়ন্ত্রণ করা যায়। এই পরবর্তী পাঠে, আইওটি ডিভাইসটি  MQTT-র মাধ্যমে একটি টেলেমেট্রি বার্তা লাইটের লেভেলসহ(সেন্সর এর ভ্যালু) পাবলিক MQTT ব্রোকারে পাঠাবে এবং  সেটি কতিপয় সার্ভার দ্বারা নেওয়া হবে যেটাতে কোডটি লেখা হয়েছিলো। এই কোডটি লাইটের লেভেল/সেন্সর ভ্যালু যাচাই করবে এবং যাচাই করার পর একটি আদেশমূলক বার্তা আইওটি লাইটটিতে/ডিভাইসে পাঠাবে যাতে আদেশ হিসেবে বলা থাকবে যে এলইডিটি অন না অফ হবে।
+
+বাস্তবিক ক্ষেত্রে এমন অবস্থা হতে পারে যেখানে অনেক লাইট সেন্সর রয়েছে (যেমন স্ট্যাডিয়াম) এবং সেই লাইটগুলো অন করার সিদ্ধান্ত নেওয়ার পূর্বে ওই একাধিক লাইট সেন্সর এর তথ্য সংগ্রহ করার প্রয়োজন হতে পারে।  শুধুমাত্র একটি সেন্সর মেঘ বা পাখি দ্বারা আবৃত থাকে লাইটগুলি অন হওয়া থেকে বন্ধ রাখতে পারে, যদিও অন্য সেন্সরগুলি পর্যাপ্ত আলো শনাক্ত করেও।
+
+✅ কমান্ড প্রেরণের আগে একাধিক সেন্সর এর তথ্য মূল্যায়নের জন্য অন্য কি পরিস্থিতিগুলো বিবেচিত হতে পারে?
+
+এসাইনমেন্টের অংশ হিসেবে MQTT ব্রোকার সেটাপের এই  জটিলতার সাথে মোকাবেলা করার চেয়ে বরং চাইলে সেটাপটি পাবলিক টেস্ট সার্ভার ব্যবহার করা যবে যেটি [Eclipse Mosquitto](https://www.mosquitto.org)(একটি ওপেন সোর্স MQTT ব্রোকার)-এ  রান হবে। এই টেস্ট ব্রোকারটি [test.mosquitto.org](https://test.mosquitto.org)-এ পাওয়া যাবে যা জনসাধারনের জন্য উন্মুক্ত। MQTT ক্লায়েন্ট এবং সার্ভার এর জন্য এটি একটি অসাধারণ টুল কারণ এটিতে সেটাপ করতে কোনো একাউন্টের প্রয়োজন নেই।
+
+> 💁 এই টেস্ট ব্রোকারটি উন্মুক্ত যা মোটেই সুরক্ষিত নয়। যে কেউ বুঝতে পারবে এতে কি পাবলিশ করা হয়েছে, তাই যে তথ্যগুলোতে গোপনীয় রাখা জরুরি সেগুলো এতে ব্যবহার না করার পরামর্শ রইল।
+
+![A flow chart of the assignment showing light levels being read and checked, and the LED begin controlled](../../../../images/assignment-1-internet-flow.png)
+
+ডিভাইসটি MQTT ব্রোকারে সংযুক্ত করতে সংশ্লিষ্ট ধাপগুলো অনুসরণ করিঃ
+
+* [আরডুইনো Wio টার্মিনাল](wio-terminal-mqtt.bn.md)
+* [সিংগেল বোর্ড কম্পিউটার - রাস্পবেরি পাই/ভার্চুয়াল আইওটি ডিভাইস](single-board-computer-mqtt.bn.md)
+
+### MQTT এর আরো গভীরে
+
+টপিকগুলোতে শ্রেণিবিন্যাস থাকতে পারে যাতে ক্লায়েন্টরা ওয়াইল্ডকার্ড ব্যবহার করে এই শ্রেণিবিন্যাসের বিভিন্ন স্তরে সাবস্ক্রাইব করতে পারে। যেমন, তাপমাত্রার টেলেমেট্রি বার্তাগুলো  `/telemetry/temperature` এই টপিকে এবং আর্দ্রতার বার্তাগুলো `/telemetry/humidity` এই টপিকে পাঠানো, তারপর ক্লাউড এপটি `/telemetry/*` এই টপিকে সাবস্ক্রাইব করে তাপমাত্রা এবং আর্দ্রতা উভয়ের টেলেমেট্রি বার্তাগুলো গ্রহণ করবে।
+
+বার্তা গ্রহণের নিশ্চয়তা প্রদান করতে বার্তাগুলো কোয়ালিটি অফ সার্ভিস(QoS) এর সাথে পাঠানো হয়।
+
+* সর্বাধিক একবার – বার্তা শুধুমাত্র একবারই পাঠানো হয় এবং ক্লায়েন্ট আর ব্রোকার বার্তাটি ডেলিভারীর প্রাপ্তি স্বীকার করতে কোনো অতিরিক্ত পদক্ষেপ নেয় না (fire and forget)।
+* অন্তত একবার – স্বীকারোক্তি গৃহীত না হওয়া পর্যন্ত বার্তা প্রেরক একাধিকবার বার্তা প্রেরণের চেষ্টা করেছিল (acknowledged delivery)।
+* ঠিক একবার – শুধু একটি বার্তা গৃহীত হয়েছে নিশ্চিত করতে প্রাপক এবং প্রেরক একটি দ্বি-স্তরের হ্যান্ডসেক এ জড়িত হয় (assured delivery)।
+
+✅ কোন পরিস্থিতিতে (fire and forget)-বার্তার পরেও একটি (assured delivery)-এর বার্তা প্রয়োজন হতে পারে?
```

Comment:
✅ কোন পরিস্থিতিতে fire and forget বার্তার পরেও একটি assured delivery এর বার্তা প্রয়োজন হতে পারে?